Articles/HowTos/CriticalBeginnerGitUsageTips.md

```diff
@@ -0,0 +1,99 @@
+For example:
+
+```
+Summary line for changes (50 chars is target but a little longer okay)
+
+More detailed explanatory text in the body (wrapped to about 72 chars).
+The blank line separating the summary from the body is critical to make
+commands like 'git log --oneline', 'git rebase -i', and others readable
+and useful.
+
+If you use an issue tracker, put references to them like:
+
+Resolves: #123
+Also see: #456, #789
+```
+
+**C) Create small local "checkpoint" commits then cleanup with 'git rebase -i'
+before pushing.**
+To provide for easy local "undos" and better organization of changes into final change-sets, commit often locally using "checkpoint" commits (convention is to use commit summary lines starting with "WIP:").
+But before pushing these commits to a remote shared branch, use [`git rebase -i @{u}`](https://www.atlassian.com/git/tutorials/rewriting-history#git-rebase-i) to clean up and reorganize the commits into good "logical" commits (see "SEPARATE CHANGES" above).
+
+**D) Always create local commits for your changes in local branches before you run any Git commands that might modify your local uncommitted changes.**
+The commands `git pull`, `git merge`, `git rebase`, or other Git operations can alter (or
+delete) your local uncommitted changes, either in the working directory or the staging
+area/index.
+So always create (sometimes temporary) commits for these before running any of these commands (unless you want to throw away uncommitted changes in your local repo, for example with `git checkout`, `git reset`, `git clean`, etc.).
+
+**E) Back up local branches every few hours of work by pushing them to some
+remote Git repo on a remote machine** in case your local machine or disk goes out.
+(This is why local Git branches are better than Git stashes because you can back them up to other repos in a version controlled way.
+You can’t do that with Git stashes.)
+
+**F) You can always recover an earlier state of any of your local branches** by
+looking at [`git reflog`](https://git-scm.com/book/en/v2/Git-Internals-Maintenance-and-Data-Recovery) and then using a combination [`git checkout`](http://marklodato.github.io/visual-git-guide/index-en.html#checkout) and/or [`git reset –hard`](http://marklodato.github.io/visual-git-guide/index-en.html#reset), etc.
+(See [How to undo (almost) anything in Git](https://github.blog/2015-06-08-how-to-undo-almost-anything-with-git/#redo-after-undo-local).)
+
+**G) Never delete a local Git repo (unless you are completely done with it).**
+Your local Git repos have a [wealth of information that can’t be pushed to other Git repos](https://www.cs.cmu.edu/~davide/howto/git_lose.html) (e.g. `git rerere` info, `git reflog` info, etc.).
+Because of **'F'**, you should never have to delete a local Git repo and re-clone to get out of some "bad" state.
```

Comment:
Here you seem to want to refer to section `F` and if you were using `###` you could use an actual inline hyperlink `#the-text-of-the-heading` instead.

Reply:
Right.  That suggests that I should convert all of these items to section headings.